server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/AbstractEndpoint.java

```diff
@@ -39,50 +50,78 @@
    * The endpoint result for the ongoing transaction.
    */
   protected String my_endpoint_result;
-
+  
   /**
-   * Indicate that an internal server error has taken place.
+   * The ASM for this endpoint, load in the `before` of the endpoint
+   * and saved in the `after`.
    */
-  public void serverError(final Response the_response, final String the_log_message) {
-    the_response.status(HttpStatus.INTERNAL_SERVER_ERROR_500);
-    Main.LOGGER.error("server error 500: " + the_log_message);
-    my_endpoint_result = the_log_message;
+  protected AbstractStateMachine my_asm;
+  
+  /**
+   * Halts the endpoint execution by ending the request and returning the
+   * most recently set response code and endpoint result.
+   */
+  protected final void halt(final Response the_response) {
+    Spark.halt(the_response.status(), my_endpoint_result);
   }
 
   /**
-   * Indicate that some data was not found.
+   * @return The classname of the ASM used by this endpoint.
    */
-  public void dataNotFound(final Response the_response, 
-                           final String the_log_message) {
-    the_response.status(HttpStatus.NOT_FOUND_404);
-    Main.LOGGER.error("server error 404: " + the_log_message);
-    my_endpoint_result = the_log_message;
+  protected abstract Class<? extends AbstractStateMachine> asmClass(); 
+  
+  /**
+   * Load the appropriate ASM from the database and make sure that
+   * the transition we wish to take is legal.
+   */
+  protected void loadAndCheckASM(final Response the_response) {
+    // get the state of the DoS dashboard
+    try {
+      my_asm = (AbstractStateMachine) (asmClass().getConstructor().newInstance());
```

Comment:
(neither will the query below)

BTW, if we're going to just construct a new one every endpoint, I guess there's no real need to store them in Main. Which may be something you also changed, but I haven't gotten to the end yet.